### PR TITLE
Make several stream manager routines public: postread_reindex, prewrite_reindex, and postwrite_reindex

### PR DESCRIPTION
**Pull request by mgduda**
The postread_reindex, prewrite_reindex, and postwrite_reindex routines in
the mpas_stream_manager module were previously private. However, code that
builds ad hoc output streams at the mpas_io_streams level that contain
indexing fields could benefit from the use of these routine to reindex
indexing fields (e.g., cellsOnCell).
    
The use of prewrite_reindex and postwrite_reindex requires some care, since
they set module variables within the mpas_stream_manager module to save and
retrieve pointers to the original indexing arrays. Problems can arise, for
example, if external code calls prewrite_reindex, then makes a call to write
output streams via the stream manager: in this case, pointers set by the
external call to prewrite_reindex will be overwritten by the internal call to
prewrite_reindex within mpas_stream_mgr_write.

Unlike the prewrite_reindex and postwrite_reindex routines, which rely on
module state, the postread_reindex routine does not rely on any module
variables in the mpas_stream_manager module, and can therefore be safely used
by external code.

### CODE REVIEW

**Review by mark-petersen — APPROVED** [review]
Passes stand-alone ocean test suite, intel and gnu. Please merge. I thought I'd review these, so we have the option to tag along with the framework a bit longer.

File: src/framework/mpas_stream_manager.F [hunk 1]
```diff
@@ -47,7 +47,9 @@ module mpas_stream_manager
               MPAS_get_stream_filename, &
               MPAS_build_stream_filename
 
-    public :: postread_reindex
+    public :: prewrite_reindex, &
+              postwrite_reindex, &
+              postread_reindex
```
Comment (MiCurry) [inline src/framework/mpas_stream_manager.F:52]: Should we make a comment here for `prewrite_reindex` and `postwrite_reindex` that says that these routines should be only be used when calling `MPAS_writeStream` and not `mpas_stream_mgr_write`? I think that would clarify to an unfamiliar user that these routines are for ad-hoc streams.

Comment (mgduda) [reply src/framework/mpas_stream_manager.F:65]: That seems like a good idea. I've pushed a new commit that adds notes in several places.

Comment (MiCurry) [reply src/framework/mpas_stream_manager.F:65]: Those notes look great!

**Review by MiCurry — APPROVED** [review]

**Pull request merged**